### PR TITLE
[ZEPPELIN-4602] Added initial version of InfluxDB interpreter

### PR DESCRIPTION
**Pull request by rhajek**
### What is this PR for?
The goal is to add support for querying InfluxDB 2.x using Flux language in Zeppelin notebook.

InfluxDB 2.0 (beta) docs
https://v2.docs.influxdata.com/v2.0/

Flux language docs:
https://docs.influxdata.com/flux/

### What type of PR is it?
Feature

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/ZEPPELIN-4602

### How should this be tested?
* First time? Setup Travis CI as described on https://zeppelin.apache.org/contribution/contributions.html#continuous-integration
* Strongly recommended: add automated unit tests for any new or changed behavior
* Outline any manual steps to test the PR here.

### Screenshots (if appropriate)
linked in docs/interpreter/influxdb.md

### CODE REVIEW

File: influxdb/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+============================================
+
+## Description:
+
+Provide InfluxDB Interpreter for Zeppelin.
+
+## Build
+
+```
+mvn -pl zeppelin-interpreter,zeppelin-display,influxdb -DskipTests package
```
Comment (zjffdu) [inline influxdb/README.md]: You can use `mvn -pl influxdb -am -DskipTests package`, influxdb depends on zeppelin-interpreter-shaded, so use `-am` would be easier. BTW, does influxdb interpreter depends on zeppelin-display ?

File: influxdb/pom.xml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        <influxdb.client.version>1.4.0</influxdb.client.version>
+        <dependency.okhttp3.version>3.13.1</dependency.okhttp3.version>
+        <dependency.gson.version>2.8.5</dependency.gson.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.influxdb</groupId>
+            <artifactId>influxdb-client-java</artifactId>
+            <version>${influxdb.client.version}</version>
```
Comment (zjffdu) [inline influxdb/pom.xml]: What is the license of this dependency ? You need to put it in LICENCE file.

File: influxdb/pom.xml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+            <version>${dependency.gson.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.squareup.okhttp3</groupId>
+            <artifactId>mockwebserver</artifactId>
+            <version>${dependency.okhttp3.version}</version>
```
Comment (zjffdu) [inline influxdb/pom.xml]: Same licence issue as above

File: influxdb/pom.xml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                <artifactId>maven-resources-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <artifactId>maven-shade-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <configuration>
+                    <skip>false</skip>
```
Comment (zjffdu) [inline influxdb/pom.xml]: It's better to enable it. We are planning to enable it for other modules as well.

File: influxdb/pom.xml (hunk 2 above)
Comment (rhajek) [reply influxdb/pom.xml]: MIT, I added license to zeppelin-distribution/src/bin_license/LICENSE

File: influxdb/pom.xml (hunk 3 above)
Comment (rhajek) [reply influxdb/pom.xml]: Apache 2, I added license to zeppelin-distribution/src/bin_license/LICENSE

File: influxdb/pom.xml (hunk 2 above)
Comment (rhajek) [reply influxdb/pom.xml]: I also added RxJava (Apache2 license) (used by InfluxDB client library)

File: influxdb/README.md (hunk 1 above)
Comment (rhajek) [reply influxdb/README.md]: The interpreter does not depends on zeppelin-display, I just copied it from another interpreter. It is very basic interpreter now. In future we would like to add syntax highlighting and code complete.

**Comment by zjffdu** [comment]
@rhajek Could you do a rebase and trigger the travis CI again ?

**Comment by rhajek** [comment]
It looks like there are some breaking changes, InfluxDBInterpreter.java:[28,39] cannot find symbol   symbol:   class BaseZeppelinContext, I will look on that.

**Comment by zjffdu** [comment]
@rhajek This is code refactoring which we just rename `BaseZeppelinContext` to `ZeppelinContext`

**Comment by zjffdu** [comment]
ping @rhajek Any update ？

**Comment by rhajek** [comment]
Oh sorry, Instead of rebase I merged 'apache:master' into my fork 'rhajek:master'.

Now my fork is synced with  'apache:master' but the git log contains also other commits :(
Are you able to merge this PR in current state, or should I create the new PR ?

**Comment by timhallinflux** [comment]
@zjffdu how does this look now?  Good to go?

**Comment by zjffdu** [comment]
@timhallinflux Could you paste the travis CI link ?

**Comment by rhajek** [comment]
> @rajeshkp Could you paste the travis CI link ?

Any updates ? Can I help with something ?

**Comment by zjffdu** [comment]
Sorry, I ping the wrong id, @rhajek  Could you rebase this PR and paste the travis CI link ?

**Comment by rhajek** [comment]
Rebased, here is the travis link: https://travis-ci.org/github/rhajek/zeppelin?utm_medium=notification&utm_source=github_status

**Comment by zjffdu** [comment]
@rhajek There's several failure, may be due to flaky test.   Could you rerun the build ?

**Comment by rhajek** [comment]
I rebased and manually tested InfluxDB interpreter in zeppelin UI. Also  `mvn -pl influxdb -am  test` is passing. Travis build is in progress.

File: docs/interpreter/influxdb.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+# InfluxDB Interpreter for Apache Zeppelin
+
+<div id="toc"></div>
+
+## Overview
+[InfluxDB](https://v2.docs.influxdata.com/v2.0/)  is an open-source time series database (TSDB) developed by InfluxData. It is written in Go and optimized for fast, high-availability storage and retrieval of time series data in fields such as operations monitoring, application metrics, Internet of Things sensor data, and real-time analytics.
+This interpreter allows to perform queries in [Flux Language](https://v2.docs.influxdata.com/v2.0/reference/flux/) in Zeppelin Notebook.
+
+### Notes
+* This interpreter is compatible with InfluxDB 2.0+ (v2 API, Flux language)
```
Comment (alexott) [inline docs/interpreter/influxdb.md:31]: According to README, the client library is compatible with 1.8 as well: https://github.com/influxdata/influxdb-client-java#influxdb-18-api-compatibility

File: influxdb/src/main/java/org/apache/zeppelin/influxdb/InfluxDBInterpreter.java [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  private static final String INFLUXDB_LOGLEVEL_PROPERTY = "influxdb.logLevel";
+
+  private static final String TABLE_MAGIC_TAG = "%table ";
+  private static final String WHITESPACE = " ";
+  private static final String NEWLINE = "\n";
+  private static final String TAB = "\t";
+  static final String EMPTY_COLUMN_VALUE = "";
+
+  InfluxDBClient client;
+  QueryApi queryApi;
```
Comment (alexott) [inline influxdb/src/main/java/org/apache/zeppelin/influxdb/InfluxDBInterpreter.java:64]: does it make sense to mark it as `volatile` ?

File: docs/interpreter/influxdb.md (hunk 5 above)
Comment (rhajek) [reply docs/interpreter/influxdb.md:31]: Thanks @alexott for the review. I will update the dependency to the latest InfluxDB client library and update README.

File: influxdb/src/main/java/org/apache/zeppelin/influxdb/InfluxDBInterpreter.java (hunk 6 above)
Comment (rhajek) [reply influxdb/src/main/java/org/apache/zeppelin/influxdb/InfluxDBInterpreter.java]: I'am not sure, I marked both variables as  `private` and `volatile`.

**Comment by rhajek** [comment]
I rebased PR, Travis also build looks ok.

**Comment by alexott** [comment]
LGTM

**Comment by zjffdu** [comment]
LGTM

**Pull request closed without merging**